scripts/quests/jeuno/Tenshodo_Membership.lua

```diff
@@ -0,0 +1,100 @@
+-----------------------------------
+-- Tenshodo Membership
+-----------------------------------
+-- Log ID: 3, Quest ID: 17
+-- NPC: Ghebi Damomohe !pos 16 -.1 -5.2 245
+-- NPC: Jabbar !pos -101 -2.3 23 236
+-- NOTE: Players can straight turn in Tenshodo Invite bypassing this quest. Quest never shows in log until completed.
+-----------------------------------
+
+local quest = Quest:new(xi.questLog.JEUNO, xi.quest.id.jeuno.TENSHODO_MEMBERSHIP)
+
+quest.reward =
+{
+    item    = xi.item.TENSHODO_INVITE,
+    keyItem = xi.ki.TENSHODO_MEMBERS_CARD,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_AVAILABLE and
+                player:getFameLevel(xi.fameArea.JEUNO) >= 3
+        end,
+
+        [xi.zone.LOWER_JEUNO] =
+        {
+            ['Ghebi_Damomohe'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, { xi.item.TENSHODO_INVITE }) then
+                        player:tradeComplete()
```

Comment:
eating the items really should happen until the event finish

Reply:
Good point. Moved.